spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -253,6 +257,39 @@ protected DeleteSummary deleteFiles(
     return summary;
   }
 
+  protected DeleteSummary deleteFiles(
```

Comment:
I am wondering if we shall have some sort of tests to cover this, where the `bulkDeleteFunc` can add files to a separate set instead of actual delete, to see we are properly group by its type and verify the DeleteSummary results?

Reply:
The ExpireSnapshots tests do this, which is why there are all of those modifications to the withBulkDeleteFunc. I think I missed a few though since there are still some failing tests. I'll check that tomorrow